### PR TITLE
Fix overwatch issue and remove older patch

### PR DESCRIPTION
**Pull request by sanyae2439**
* Transpiler patch
* Testing: OK (on 2 players)

This patch ignores Overwatch players on first assign roles.

* Removed older patch, its my PR patch (#94)
(currently this patch does nothing and returns original CodeInstruction)

### CODE REVIEW

File: Exiled.Events/Patches/Fixes/OverwatchSkipFix.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            // if (referenceHub.serverRoles.OverwatchEnabled)
+            // {
+            //      list.Add(RoleType.Spectator);
+            //      playersRoleList.Add(referenceHub.gameObject, RoleType.Spectator);
+            //      continue;
+            // }
+            newInstructions.InsertRange(index, new[]
+            {
+                // if (referenceHub.serverRoles.OverwatchEnabled)
+                //      continue;
```
Comment (PintTheDragon) [inline Exiled.Events/Patches/Fixes/OverwatchSkipFix.cs]: These comments seem to contradict each other. I would remove the `//      continue;` from the second one, and add it to where the continue actually is.

Comment (PintTheDragon) [reply Exiled.Events/Patches/Fixes/OverwatchSkipFix.cs]: Oh lmao looks like this is already fixed. Well, that was the only issue I saw.

File: Exiled.Events/Patches/Fixes/OverwatchSkipFix.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    using HarmonyLib;
+
+    using NorthwoodLib.Pools;
+
+    using UnityEngine;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Fixes <see cref="CharacterClassManager.RunDefaultClassPicker(bool, out int[], out System.Collections.Generic.Dictionary{UnityEngine.GameObject, RoleType})"/> method.
```
Comment (PintTheDragon) [inline Exiled.Events/Patches/Fixes/OverwatchSkipFix.cs:24]: This should probably be changed to something like ```
Patches <see cref="CharacterClassManager.RunDefaultClassPicker(bool, out int[], out System.Collections.Generic.Dictionary{UnityEngine.GameObject, RoleType})"/>.
Fixes overwatch players not spawning correctly.```

Comment (sanyae2439) [reply Exiled.Events/Patches/Fixes/OverwatchSkipFix.cs:24]: add comments.

File: Exiled.Events/Patches/Fixes/OverwatchSkipFix.cs [hunk 3]
```diff
@@ -22,6 +22,7 @@ namespace Exiled.Events.Patches.Fixes
 
     /// <summary>
     /// Fixes <see cref="CharacterClassManager.RunDefaultClassPicker(bool, out int[], out System.Collections.Generic.Dictionary{UnityEngine.GameObject, RoleType})"/> method.
```
Comment (PintTheDragon) [inline Exiled.Events/Patches/Fixes/OverwatchSkipFix.cs:24]: The top line still needs to be changed. `Patches <see cref="CharacterClassManager.RunDefaultClassPicker(bool, out int[], out System.Collections.Generic.Dictionary{UnityEngine.GameObject, RoleType})"/>.` should work.

**Review by PintTheDragon — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Fixes/OverwatchSkipFix.cs (hunk 3 above)
Comment (sanyae2439) [reply Exiled.Events/Patches/Fixes/OverwatchSkipFix.cs]: oof, i fixed again

**Review by PintTheDragon — APPROVED** [review]
Looks good!

**Review by iRebbok — APPROVED** [review]

**Comment by iRebbok** [comment]
Left the `base-game-bug` label, not sure if that's on our side.

**Comment by sanyae2439** [comment]
this is a main game bug, and old workaround wasnt good.

**Review by iopietro — APPROVED** [review]

**Pull request merged**